Refit/IErrorHandler.cs

```diff
@@ -0,0 +1,14 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Net.Http;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Refit
+{
+    public interface IErrorHandler
+    {
+        Task<Exception> HandleErrorAsync(HttpRequestMessage message, HttpMethod httpMethod, HttpResponseMessage response,
+            RefitSettings refitSettings = null);
```

Comment:
`refitSettings` shoudln't default to `null` as it's called by library code, which we control. `refitSettings` should never be null.

Reply:
This is due to the fact that in `RestMethodInfo` constructor it may be null https://github.com/reactiveui/refit/pull/927/files#diff-12d38083c42706a6ad12f6b11083cea6R43.
Should i do refitSettings not null in `IErrorHandler` and `RestMethodInfo`? This will require changing tests in `Refit.Tests\RequestBuilder.cs`